packages/benchmarks/src/textToDriver/nodeJsDriverBenchmark.ts

```diff
@@ -1,98 +1,139 @@
 import { runTextToDriverEval } from "./runTextToDriverEval";
-import { radiantModels } from "../radiantModels";
-import { assertEnvVars, MongoClient } from "mongodb-rag-core";
+import { models } from "../models";
+import {
+  assertEnvVars,
+  CORE_OPENAI_CONNECTION_ENV_VARS,
+  MongoClient,
+} from "mongodb-rag-core";
 import { NODE_JS_PROMPTS } from "./generateDriverCode/languagePrompts/nodeJs";
-import OpenAI from "openai";
 import { TEXT_TO_DRIVER_ENV_VARS } from "./TextToDriverEnvVars";
-import { RADIANT_ENV_VARS } from "../envVars";
+import { BRAINTRUST_ENV_VARS, RADIANT_ENV_VARS } from "../envVars";
 import { wrapOpenAI } from "braintrust";
 import PromisePool from "@supercharge/promise-pool";
+import { makeOpenAiClientFactory } from "../makeOpenAiClientFactory";
 async function sleep(ms: number) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 async function main() {
   const {
     BRAINTRUST_API_KEY,
     BRAINTRUST_TEXT_TO_DRIVER_PROJECT_NAME,
+    BRAINTRUST_ENDPOINT,
     MONGODB_TEXT_TO_DRIVER_CONNECTION_URI,
     RADIANT_API_KEY,
     RADIANT_ENDPOINT,
     MONGODB_AUTH_COOKIE,
-  } = assertEnvVars({ ...TEXT_TO_DRIVER_ENV_VARS, ...RADIANT_ENV_VARS });
+    OPENAI_API_KEY,
+    OPENAI_ENDPOINT,
+    OPENAI_API_VERSION,
+  } = assertEnvVars({
+    ...TEXT_TO_DRIVER_ENV_VARS,
+    ...RADIANT_ENV_VARS,
+    ...BRAINTRUST_ENV_VARS,
+    ...CORE_OPENAI_CONNECTION_ENV_VARS,
+  });
   const projectName = BRAINTRUST_TEXT_TO_DRIVER_PROJECT_NAME;
   const datasetName = "text-to-query-results";
-  const maxConcurrency = 3;
-  const timeout = 60000;
+  const DEFAULT_MAX_CONCURRENCY = 3;
+  const openAiClientFactory = makeOpenAiClientFactory({
+    azure: {
+      apiKey: OPENAI_API_KEY,
+      apiVersion: OPENAI_API_VERSION,
+      endpoint: OPENAI_ENDPOINT,
+    },
+    radiant: {
+      apiKey: RADIANT_API_KEY,
+      endpoint: RADIANT_ENDPOINT,
+      authCookie: MONGODB_AUTH_COOKIE,
+    },
+    braintrust: {
+      apiKey: BRAINTRUST_API_KEY,
+      endpoint: BRAINTRUST_ENDPOINT,
+    },
+  });
 
   const prompts = NODE_JS_PROMPTS.systemPrompts;
   const mongoClient = new MongoClient(MONGODB_TEXT_TO_DRIVER_CONNECTION_URI);
 
   const SAMPLE_DOCUMENT_LIMIT = 2;
 
+  const { RUN_ID } = process.env;
+
   try {
     await mongoClient.connect();
     await sleep(500);
-    const modelExperiments = radiantModels.map((modelInfo) => {
-      const modelExperiments = [];
-      for (const promptType of Object.keys(prompts)) {
-        for (const generateCollectionSchemas of [true, false]) {
-          modelExperiments.push({
-            modelInfo,
-            promptType,
-            generateCollectionSchemas,
-          });
+    const modelExperiments = models
+
+      .filter((m) => m.authorized === true)
+      // // NOTE: ignoring Google models for now b/c of issues with Radiant
+      .filter((m) => m.developer !== "Google")
+      // TODO: sequentially work thru the models
+      .filter((m) => m.label.includes("llama"))
```

Comment:
Why do we only run if there is a `llama` label?

Also, is the TODO done or is there an EAI we can add to the comment?

Reply:
ah this line can be removed. this was just for me to do while iterating thru

Reply:
```suggestion
```